### PR TITLE
[Python 3] Automaton fixes & __next__ python 3 alias

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- Fixes small forgotten bugs in the automaton system
- Add `__next__` functions as `next` alias for python 3

### CODE REVIEW

**Comment by gpotter2** [comment]
@p-l- This should fix some Appveyor tests.
Small PR, pretty safe to merge.

File: scapy/automaton.py [hunk 1]
```diff
@@ -86,16 +91,17 @@ def call_release(self, arborted=False):
         self.was_ended = arborted
         try:
             self.trigger.release()
-        except thread.error:
-            pass
+        except Exception as e:
+            if not isinstance(e, RuntimeError) and not (six.PY2 and isinstance(e, thread.error)):
```
Comment (gpotter2) [inline scapy/automaton.py]: On python 2, the error is thread.error, on python 3 it's RuntimeError

File: scapy/automaton.py [hunk 2]
```diff
@@ -21,6 +21,11 @@
 from scapy.consts import WINDOWS
 import scapy.modules.six as six
 
+try:
```
Comment (p-l-) [inline scapy/automaton.py]: Here is what I propose:
```Python
try:
    import thread
except ImportError:
    THREAD_EXCEPTION = RuntimeError
else:
    THREAD_EXCEPTION = thread.error
```

Then
```Python
        try:
            self.trigger.release()
        except THREAD_EXCEPTION:
            pass
```

File: scapy/automaton.py [hunk 3]
```diff
@@ -625,8 +634,10 @@ def __init__(self, *args, **kargs):
                 ioin = ObjectPipe()
             elif not isinstance(ioin, types.InstanceType):
                 ioin = self._IO_fdwrapper(ioin,None)
-            if ioout is None:
+            if ioout is None and not WINDOWS:
```
Comment (p-l-) [inline scapy/automaton.py]: Please rewrite this as:
```Python
            if ioout is None:
                ioout = ioin if WINDOWS else ObjectPipe()
```

File: scapy/automaton.py (hunk 2 above)
Comment (gpotter2) [inline scapy/automaton.py]: Done

**Pull request merged**